warp10/src/main/java/io/warp10/WarpDist.java

```diff
@@ -168,46 +169,61 @@ public static void main(String[] args) throws Exception {
     //
     // Set SIPHASH keys for class/labels/index
     //
-    
+
     for (String property: REQUIRED_PROPERTIES) {
       Preconditions.checkNotNull(properties.getProperty(property), "Property '" + property + "' MUST be set.");
     }
 
     //
-    // Decode generic keys
+    // Decode generic keys and secrets
     // We do that first so those keys do not have precedence over the specific
     // keys.
     //
-    
+
+    Map<String,String> secrets = new HashMap<String,String>();
     for (Entry<Object,Object> entry: properties.entrySet()) {
       if (entry.getKey().toString().startsWith(Configuration.WARP_KEY_PREFIX)) {
         byte[] key = keystore.decodeKey(entry.getValue().toString());
         if (null == key) {
           throw new RuntimeException("Unable to decode key '" + entry.getKey() + "'.");
         }
         keystore.setKey(entry.getKey().toString().substring(Configuration.WARP_KEY_PREFIX.length()), key);
+        // Remove the key definition
+        properties.remove(entry.getKey());
+      } else if (entry.getKey().toString().startsWith(Configuration.WARP_SECRET_PREFIX)) {
+        byte[] secret = keystore.decodeKey(entry.getValue().toString());
+        if (null == secret) {
+          throw new RuntimeException("Unable to decode secret '" + entry.getKey() + "'.");
+        }
+        // Encode secret as OPB64 and store it
+        secrets.put(entry.getKey().toString().substring(Configuration.WARP_SECRET_PREFIX.length()), OrderPreservingBase64.encodeToString(secret));
+        // Remove the secret from the configuraiton
+        properties.remove(entry.getKey());
```

Comment:
Won't it raise a `ConcurrentModificationException` or lead to an unwanted behavior?
Same line 192, and lines 188 and 197 in Warp.java.